File: p3226/cpp/main.cpp
#include <iostream>
#include <bit>

class Solution {
public:
    int minChanges(int n, int k) {
        if ((n & k) !=k)
        {
            return -1;
        }
        int nuevo_value = (n ^k);
        return std::popcount(static_cast<unsigned int>(nuevo_value));
    }
};

File: p3226/rust/src/main.rs
struct Solution;

impl Solution {
    pub fn min_changes(n: i32, k: i32) -> i32 {
        if n == k {
            return 0;
        }
        if (n & k) !=k {
            return -1;
        }
        (n ^ k).count_ones() as i32
    }
}
fn main() {
    println!("Hello, world!");
}
